### PR TITLE
Support OpenCL.DebugInfo.100 extended instruction set

### PR DESCRIPTION
**Pull request by dneto0**
The interesting bits are:
- The extended instruction set adds enum types, with names that overlap with enum types from  the older DebugInfo extension.  This affects spv_operand_type_t enum values.
- So I added a mechanism to prefix the operand-kind names, so distinguish the cases. 
- The enums from the old DebugInfo are left unprefixed, but we do add a CLDEBUG100_ prefix to the new ones.

### CODE REVIEW

**Comment by dneto0** [comment]
The OpenCL.DebugInfo.100 extended instruction has been ratified by Khronos but is not yet on the SPIR-V registry.

**Comment by dneto0** [comment]
cc: @AlexeySotkin

**Comment by dneto0** [comment]
I don't have an active GN setup, so I took a guess at the BUILD.gn changes.   Cc: @dj2 @zoddicus

**Comment by AlexeySotkin** [comment]
:thumbsup: @dneto0 Thank you very much!

**Comment by ehsannas** [comment]
Is there going to be a header file for the opcodes of this instruction set?
(similar to [this](https://github.com/KhronosGroup/SPIRV-Headers/blob/master/include/spirv/unified1/GLSL.std.450.h) for the GLSL ext set.)

File: source/extinst.opencl.debuginfo.100.grammar.json [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+      "opcode" : 0
+    },
+    {
+      "opname" : "DebugCompilationUnit",
+      "opcode" : 1,
+      "operands" : [
+        { "kind" : "IdRef", "name" : "'Source'" },
+        { "kind" : "LiteralInteger", "name" : "'Version'" },
+        { "kind" : "LiteralInteger", "name" : "'DWARF Version'" }
+      ]
```
Comment (dneto0) [inline source/extinst.opencl.debuginfo.100.grammar.json]: This is missing some fields, (thanks @jaebaek for finding this)

**Comment by dneto0** [comment]
Hey @jaebaek please take a look now.  I think I fixed the operands to all the instructions now.

File: source/extinst.opencl.debuginfo.100.grammar.json [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+        { "kind" : "IdRef", "name" : "'Source'" },
+        { "kind" : "LiteralInteger", "name" : "'Line'" },
+        { "kind" : "LiteralInteger", "name" : "'Column'" },
+        { "kind" : "IdRef", "name" : "'Parent'" }
+      ]
+    },
+    {
+      "opname" : "DebugTypeFunction",
+      "opcode" : 8,
+      "operands" : [
```
Comment (jaebaek) [inline source/extinst.opencl.debuginfo.100.grammar.json]: The first operand must be `DebugInfoFlags`

File: source/extinst.opencl.debuginfo.100.grammar.json [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+    {
+      "opname" : "DebugTypeComposite",
+      "opcode" : 10,
+      "operands" : [
+        { "kind" : "IdRef", "name" : "'Name'" },
+        { "kind" : "DebugCompositeType", "name" : "'Tag'" },
+        { "kind" : "IdRef", "name" : "'Source'" },
+        { "kind" : "LiteralInteger", "name" : "'Line'" },
+        { "kind" : "LiteralInteger", "name" : "'Column'" },
+        { "kind" : "IdRef", "name" : "'Parent'" },
```
Comment (jaebaek) [inline source/extinst.opencl.debuginfo.100.grammar.json]: `{ "kind" : "IdRef", "name" : "'Linkage Name'" },` is missing

File: source/extinst.opencl.debuginfo.100.grammar.json [hunk 4]
```diff
@@ -0,0 +1,627 @@
...
+    {
+      "opname" : "DebugLocalVariable",
+      "opcode" : 26,
+      "operands" : [
+        { "kind" : "IdRef", "name" : "'Name'" },
+        { "kind" : "IdRef", "name" : "'Type'" },
+        { "kind" : "IdRef", "name" : "'Source'" },
+        { "kind" : "LiteralInteger", "name" : "'Line'" },
+        { "kind" : "LiteralInteger", "name" : "'Column'" },
+        { "kind" : "IdRef", "name" : "'Parent'" },
```
Comment (jaebaek) [inline source/extinst.opencl.debuginfo.100.grammar.json]: `{ "kind" : "DebugInfoFlags", "name" : "'Flags'" }` is missing

File: source/extinst.opencl.debuginfo.100.grammar.json [hunk 5]
```diff
@@ -0,0 +1,627 @@
...
+      "operands" : [
+        { "kind" : "IdRef", "name" : "'Local Variable'" },
+        { "kind" : "IdRef", "name" : "'Variable'" },
+        { "kind" : "IdRef", "name" : "'Expression'" }
+      ]
+    },
+    {
+      "opname" : "DebugValue",
+      "opcode" : 29,
+      "operands" : [
```
Comment (jaebaek) [inline source/extinst.opencl.debuginfo.100.grammar.json]: `{ "kind" : "IdRef", "name" : "'Local Variable'" },` is missing.

**Comment by dneto0** [comment]
> Is there going to be a header file for the opcodes of this instruction set?

The build generates an OpenCLDebugInfo.h file in the build output directly.
Ideally all these grammar files would be migrated into the KhronosGroup/SPIRV-Headers repo, and the tooling to generate headers too.

**Comment by dneto0** [comment]
The generated header file is at this gist:  https://gist.github.com/dneto0/a8fe7e0362b817092fde746a97322204

**Comment by ehsannas** [comment]
SG. Thanks!

**Comment by dneto0** [comment]
Hi @jaebaek  I think you reviewed the old commit.  I posted another commit after that. Please take another look.

**Comment by jaebaek** [comment]
Oh, you are right. It looks good to me now.

**Comment by dneto0** [comment]
Rebased. Keeping the approval.

**Pull request merged**